lib/perl/centreon/script/centcore.pm

```diff
@@ -439,116 +437,123 @@ sub removeIllegalCharacters($) {
-                    $cmd2 = "$self->{echo} \"".$cmd_line."\" >> ".$command_file;
-                    $self->{logger}->writeLogInfo("External command on Central Server: ($id) : \"".$cmd_line."\"");
-                    ($lerror, $stdout, $return_code) = centreon::common::misc::backtick(
-                        command => $cmd2,
-                        logger => $self->{logger},
-                        timeout => $self->{cmd_timeout}
-                    );
-                    $cmd_line = "";
-                    $count = 0;
-                }
-                if ($lerror != 0 || (defined($return_code)  && $return_code != 0)) {
-                    $self->{logger}->writeLogError(
-                        "Could not write into pipe file " . $command_file . " on poller " . $id
-                    );
                 }
             } else {
                 $self->{logger}->writeLogError(
-                    'Cannot write external command on central server : "' . $cmd_line . '"'
+                    "Cannot write external command on local server. File '" . $commandFile . "' not available."
                 );
             }
         } else {
-            $cmd =~ s/\'/\'\\\'\'/g;
-
-            # split $cmd in order to send it in multiple line
+            my $remoteServer = '';
+            my $multipleCommands = '';
             my $count = 0;
             my $totalCount = 0;
-            my @splittedCommands = split(/\n/, $cmd);
-            my $countCommands = @splittedCommands;
-            foreach my $cmd1 (@splittedCommands) {
-                $cmd_line .= $self->removeIllegalCharacters($cmd1) . "\n";
+            foreach my $command (@{$commandArray}) {
+                $command =~ s/\'/\'\\\'\'/g;
+                $command = $self->removeIllegalCharacters($command);
                 $count++;
                 $totalCount++;
 
-                if ($count >= 200 || $totalCount == $countCommands) {
-                    if (defined($server_info->{remote_id}) && $server_info->{remote_id} != 0 && $self->{instance_mode} ne "remote") {
-                        my $remote_server = $self->getServerConfig($server_info->{remote_id});
-                        $cmd_line =~ s/^\s+|\s+$//g;
-                        $cmd2 = "$self->{ssh} -q " . $remote_server->{ns_ip_address} . " -p $port "
-                            . "\"$self->{echo} 'EXTERNALCMD:$id:" . $cmd_line . "' "
+                if (defined($serverInfo->{remote_id}) && $serverInfo->{remote_id} != 0
+                    && $self->{instance_mode} ne "remote" && $serverInfo->{remote_server_centcore_ssh_proxy} == 1) {
+                    $remoteServer = $self->getServerConfig($serverInfo->{remote_id});
+                    # Add Centcore MACRO when routing to Remote Server
+                    $command = "EXTERNALCMD:" . $pollerId . ":" . $command;
+                    $multipleCommands .= $command . "\n";
+                } else {
+                    $multipleCommands .= $command . "\n";
```

Comment:
this line could be put outside of the instruction, avoiding to put it in "if" and "else"

Reply:
Indeed.